plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/getbean/DetachableBeanHolder.java

```diff
@@ -127,10 +133,11 @@ public Object getBean() throws IllegalAccessException, InvocationTargetException
         if (beanCopy == null) {
             Method[] methods = beanFactory.getClass().getMethods();
             for (Method factoryMethod : methods) {
+                Class<?>[] parameterTypes = factoryMethod.getParameterTypes();
                 if (ProxyReplacer.FACTORY_METHOD_NAME.equals(factoryMethod.getName())
-                        && Arrays.equals(factoryMethod.getParameterTypes(), paramClasses)) {
+                        && parameterTypes.length == 1 && parameterTypes[0] == String.class) {
```

Comment:
One last question, If we use beanFactory.getBean(xxx.class), and the name of this bean is changed, it will cannot get the right bean.   Maybe we should store the getBean method and args.

I have no other question.

Reply:
You are right, but now we are enhancing `createBean` instead of `getBean`, so it is not so easy to get the parameters of `getBean`.

Regarding the way to get parameters, what I can think of is to enhance `getBean`, store the method parameters in `ThreadLocal`, and take them out when creating `DetachableBeanHolder`. What do you think? @cvictory @beiwei30